test/olm/component_test.go

```diff
@@ -0,0 +1,94 @@
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	"github.com/redhat-developer/devopsconsole-operator/pkg/apis"
+	componentsv1alpha1 "github.com/redhat-developer/devopsconsole-operator/pkg/apis/devopsconsole/v1alpha1"
+
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+)
+
+// ComponentTest does e2e test as per operator-sdk documentation
+// https://github.com/operator-framework/operator-sdk/blob/cc7b175/doc/test-framework/writing-e2e-tests.md
+func TestComponent(t *testing.T) {
+	var err error
+	// Register types with framework scheme
+	componentList := &componentsv1alpha1.ComponentList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Component",
+			APIVersion: "devopsconsole.openshift.io/v1alpha1",
+		},
+	}
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, componentList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	t.Log(fmt.Sprintf("namespace: %s", namespace))
+	// wait for component-operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, "operators", "devopsconsole-operator", 1, retryInterval, timeout*2)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("component-operator is ready and running state")
+
+	// create a Component custom resource
+	cr := &componentsv1alpha1.Component{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Component",
+			APIVersion: "devopsconsole.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mycomp",
+			Namespace: namespace,
+		},
+		Spec: componentsv1alpha1.ComponentSpec{
+			BuildType: "nodejs",
+			Codebase:  "https://github.com/nodeshift-starters/nodejs-rest-http-crud",
+		},
+	}
+	// use TestCtx's create helper to create the object and add a cleanup function for the new object
+	err = f.Client.Create(context.TODO(), cr, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to create custom resource of kind `Component`")
+
+	t.Run("retrieve component and verify related resources are created", func(t *testing.T) {
```

Comment:
I think this is very much similar to what we have in `test/e2e` package only one test folder should be kept. It make sense that's the olm test cover the whole e2e. I'd suggest merging the content of the other test in here. In the other file we do 3 test cases:
- one with openshift nodejs repo not present then component CR need to fetch from docker image stream
- one with all date and openshift nodejs image stream (happy path)
- one with missing required attributes.
@Avni-Sharma @tinakurian wdyt on merging those tests?

Reply:
We are trying to have a way to run two kinds of test jobs/make-targets
1. Test the entire lifecycle using OLM
2. Test only the operator behavior.